### PR TITLE
lib(table): Make sure that the expandable part of table rows has unique key

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

File: pkg/lib/cockpit-components-table.jsx [hunk 1]
```diff
@@ -170,7 +170,8 @@ export class ListingTable extends React.Component {
                     parent: rowIndex - 1,
                     cells: [{ title: currentValue.expandedContent }],
                     fullWidth: true, noPadding: true,
-                    rowId: currentValue.rowId ? cockpit.format("$0-expanded", currentValue.rowId) : undefined
+                    rowId: currentValue.rowId ? cockpit.format("$0-expanded", currentValue.rowId) : undefined,
+                    props: { key: cockpit.format("$0-expanded", currentValue.props.key) },
```
Comment (martinpitt) [inline pkg/lib/cockpit-components-table.jsx:174]: It's a bit expensive to use cockpit.format for that -- a simple `+` would certainly do. :-)

File: pkg/lib/cockpit-components-table.jsx [hunk 2]
```diff
@@ -247,6 +248,7 @@ ListingTable.propTypes = {
     caption: PropTypes.string,
     emptyCaption: PropTypes.node,
     columns: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.object, PropTypes.string])),
+    props: PropTypes.object.isRequired,
```
Comment (martinpitt) [inline pkg/lib/cockpit-components-table.jsx:251]: I don't understand this commit -- This introduces a new required property, but does not update any of the existing code to add it. Also, "props" is very generic -- everything is a property, and IMHO they should be flat, not indirect like this. The only allowed (and expected) key in it is `key` right now, right?

Comment (KKoukiou) [reply pkg/lib/cockpit-components-table.jsx:251]: That was a mistake actually - props is a mandatory key inside rows. Sorry for confusion.

Comment (KKoukiou) [reply pkg/lib/cockpit-components-table.jsx]: This only mandatory is the row = > { props: { key: THIS IS MANDATORY} }
See https://github.com/cockpit-project/cockpit/pull/14447/files#diff-817714eadf61479bc760742bcca467a8R46

**Comment by martinpitt** [comment]
/home/user/cockpit-project/bots/make-checkout-workdir/pkg/lib/cockpit-components-table.jsx
      20:8  error  'cockpit' is defined but never used  no-unused-vars

**Comment by KKoukiou** [comment]
> ```
> /home/user/cockpit-project/bots/make-checkout-workdir/pkg/lib/cockpit-components-table.jsx
>   20:8  error  'cockpit' is defined but never used  no-unused-vars
> ```

Ups sorry - I should never we so confident to not compile obviously.

**Pull request merged**